prov/efa/src/efa_hmem.c

```diff
@@ -137,9 +137,7 @@ static int efa_hmem_info_init_cuda(struct efa_hmem_info *cuda_info, struct efa_d
 	}
 
 	cuda_info->p2p_disabled_by_user = false;
-
-	/* TODO set to false when non-p2p support is added */
-	cuda_info->p2p_required_by_impl = true;
+	cuda_info->p2p_required_by_impl = false;
```

Comment:
This is not quite right. We still need to check `cuda_get_xfer_setting()`. Only when `cuda_get_xfer_setting()` is `CUDA_XFER_ENABLED` we do not require p2p by implemenation.

Reply:
Based on your changes in #8357, it would have to be `FI_HMEM_CUDA_ENABLE_XFER=1` or `FI_HMEM_CUDA_USE_GDRCOPY=1`. Otherwise there'd be no path to use GDRCopy.

Reply:
We do not want to use gdrcopy to support non-p2p.